lib/dialogs/import/importers.ts

```diff
@@ -26,4 +28,46 @@ const importers: Array<importer> = [
   },
 ];
 
-export default importers;
+export const getImporters = () => {
+  return importers;
+};
+
+export const forFilename = (file: String): Importer => {
+  const fileExtension =
+    file.substring(file.lastIndexOf('.') + 1, file.length) || file;
+
+  switch (fileExtension) {
+    case 'json':
+      return importers.find((importer) => importer.name === 'simplenote');
+
+    case 'enex':
+      return importers.find((importer) => importer.name === 'evernote');
+
+    case 'txt':
+    case 'md':
+      return importers.find((importer) => importer.name === 'text-files');
+  }
+
+  throw new Error(`No importer found for file ${file}`);
+};
+
+export const load = (name: ImporterName): Promise<object> => {
```

Comment:
it's probably worth putting an explanation here - "here be dragons" - to warn people about the dynamic imports. it doesn't need to be much

```ts
/** Async-loads importer JS bundle
 *
 * Warning! Don't replace the static strings with string-interpolation
 * or else they won't be as easy to find and web pack will generate
 * more than we expect.
 */
```

Reply:
Good idea!